salt/client/ssh/wrapper/state.py

```diff
@@ -981,7 +981,7 @@ def show_sls(mods, saltenv='base', test=None, **kwargs):
             __context__['fileclient'])
     st_.push_active()
     if isinstance(mods, six.string_types):
-        mods = mods.split(',')
+        mods = [item.strip() for item in mods.split(',')]
```

Comment:
I would actually refactor this a bit. Above my suggestion with `filter` is OK-ish, but since we already have all that thrice, I would rather refactor it removing repetitive code in the following way:

```python
def _parse_mods(mods):
    '''
    Parse modules.
    '''
    if isinstance(mods, six.string_types):
        mods = filter(None, [item.strip() for item in mods.split(',')]

    return mods

...
    mods = _parse_mods(mods)  # everywhere
```

Reply:
Moved to method. 
Calling method plus cleaning the string check since its already inside the _parse_mods